frontend/src/container/GridGraphLayout/Graph/FullView/GraphManager.tsx

```diff
@@ -0,0 +1,242 @@
+			true,
+		);
+		setTableDataSet(getDefaultTableDataSet(data));
+		data.datasets.forEach((d, i) => {
+			const index = legendEntries.findIndex((di) => di.label === d.label);
+			if (index !== -1) {
+				newGraphVisibilityStates[i] = legendEntries[index].show;
+			}
+		});
+		setGraphVisibilityState(newGraphVisibilityStates);
+	}, [data, legendEntries]);
+
+	useEffect(() => {
+		if (localStorage.getItem(LOCALSTORAGE.GRAPH_VISIBILITY_STATES) !== null) {
+			const legendGraphFromLocalStore = localStorage.getItem(
+				LOCALSTORAGE.GRAPH_VISIBILITY_STATES,
+			);
+			const legendFromLocalStore: [
+				{ name: string; dataIndex: LegendEntryProps[] },
+			] = JSON.parse(legendGraphFromLocalStore || '[]');
+			let isfound = false;
+			const newGraphVisibilityStates = Array(data.datasets.length).fill(true);
+			legendFromLocalStore.forEach((item) => {
+				// if the legend entries are found in the local storage
+				if (item.name === name) {
+					setLegendEntries(item.dataIndex);
+					data.datasets.forEach((dataset, i) => {
+						const index = item.dataIndex.findIndex(
+							(dataKey) => dataKey.label === dataset.label,
+						);
+						if (index !== -1) {
+							newGraphVisibilityStates[i] = item.dataIndex[index].show;
+						}
+					});
+					setGraphVisibilityState(newGraphVisibilityStates);
+					isfound = true;
+				}
+			});
+			// if the legend entries are not found in the local storage
+			if (!isfound) {
+				setGraphVisibilityState(Array(data.datasets.length).fill(true));
+				setLegendEntries(showAllDataSet(data));
+			}
+		} else {
+			// if the legend entries are not found in the local storage
+			setGraphVisibilityState(Array(data.datasets.length).fill(true));
+			setLegendEntries(showAllDataSet(data));
+		}
+	}, [data, name]);
+
+	const checkBoxOnChangeHandler = (
+		e: CheckboxChangeEvent,
+		index: number,
+	): void => {
+		graphVisibilityState[index] = e.target.checked;
+		setGraphVisibilityState([...graphVisibilityState]);
+		if (graphVisibilityStateHandler) {
+			graphVisibilityStateHandler(graphVisibilityState);
+		}
+	};
+
+	const getCheckBox = (index: number): React.ReactElement => (
+		<CheckBox
+			data={data}
+			index={index}
+			graphVisibilityState={graphVisibilityState}
+			checkBoxOnChangeHandler={checkBoxOnChangeHandler}
+		/>
+	);
+
+	const labelClickedHandler = (labelIndex: number): void => {
+		const newGraphVisibilityStates = Array(data.datasets.length).fill(false);
+		newGraphVisibilityStates[labelIndex] = true;
+		setGraphVisibilityState([...newGraphVisibilityStates]);
+		if (graphVisibilityStateHandler) {
+			graphVisibilityStateHandler(newGraphVisibilityStates);
+		}
+	};
+
+	const getLabel = (label: string, labelIndex: number): React.ReactElement => (
+		<LabelContainer
+			type="button"
+			onClick={(): void => labelClickedHandler(labelIndex)}
+		>
+			{getAbbreviatedLabel(label)}
+		</LabelContainer>
+	);
+
+	const columns: ColumnType<DataSetProps>[] = [
```

Comment:
somewhere we need to handle the logic where we are mapping the component to scale this one

Reply:
We will do this as a part of new PR of all the Table in application